src/epik_eval/__main__.py

```diff
@@ -0,0 +1,483 @@
+            if accelerator.is_main_process:
+                if accuracies['val'] > best_val_accuracy:
+                    best_val_accuracy = accuracies['val']
+                    selected_train_accuracy = accuracies['train']
+                    selected_test_accuracy = accuracies['test']
+                    selected_test_acc_per_task = accuracies['test_per_task']
+                    best_epoch = epoch
+                tqdm_bar.set_postfix({'train loss': train_loss,
+                                      'train accuracy': accuracies['train'],
+                                      'val accuracy': accuracies['val']})
+                if args.wandb:
+                    accelerator.log({'Train Loss': train_loss,
+                                     'Train Accuracy (Memorization)': accuracies['train'],
+                                     'Val Accuracy': accuracies['val']} | accuracies['val_per_task'],
+                                    step=epoch)
+        elif accelerator.is_main_process:
+            tqdm_bar.set_postfix({'train loss': train_loss})
+            if args.wandb:
+                accelerator.log({'Train Loss': train_loss}, step=epoch)
+
+        if (epoch % args.checkpoint_every) == 0:
+            _save_checkpoint(
+                args.checkpoint_directory,
+                accelerator,
+                best_val_accuracy,
+                selected_train_accuracy,
+                selected_test_accuracy,
+                selected_test_acc_per_task,
+                best_epoch,
+                epoch,
+                elapsed_time +
+                time.time() -
+                start_time)
+
+    if accelerator.is_main_process and args.wandb:
+        accelerator.log({'Test Accuracy': selected_test_accuracy} | selected_test_acc_per_task, step=epoch)
+        accelerator.end_training()  # Tells trackers such as wandb to end
+    accelerator.print(
+        f'Benchmark finished.\n'
+        f'Elapsed time: {(elapsed_time + time.time() - start_time)/60:.2f} minutes\n'
+        f'Test Accuracy: {selected_test_accuracy:.2f}% (Epoch {best_epoch}, Train Accuracy {selected_train_accuracy:.2f}%, Val Accuracy {best_val_accuracy:.2f}%)\n'
+        f'Model answers recorded in: {args.model_answers_csv.replace(".csv", "_epoch%.csv")}')
+
+
+def _train_epoch(
+        accelerator,
+        model,
+        train_dataloader,
+        optimizer,
+        lr_scheduler,
+        criterion,
+        device,
+        args):
+
+    model.train()
+
+    # Epoch metrics
+    epoch_loss = 0
+    epoch_samples = 0
+
+    for batch in iter(train_dataloader):
+        if args.model_type == 'causal':
+            batch, attention_mask, seq_length, _ = batch
+        elif args.model_type == 't5':
+            batch, attention_mask, seq_length, target, target_length, _ = batch
+
+        # Iterate minibatches
+        for i in range(0, batch.shape[0], args.minibatch_size):
```

Comment:
The minibatch size is per device. If I were to initialize the dataloader with the minibatch size instead of the total batch size, it would divide the minibatch size by the number of device, which would change how I defined it and thus its value in each config.